beets/importer.py

```diff
@@ -811,55 +811,74 @@ def reimport_metadata(self, lib):
         """For reimports, preserves metadata for reimported items and
         albums.
         """
+        # Album and item flex attrs that should not be preserved on reimports.
+        dont_preserve_album_flex = ['data_source']
+        dont_preserve_item_flex = ['data_source', 'bandcamp_album_id',
+                                   'spotify_album_id', 'deezer_album_id',
+                                   'beatport_album_id']
```

Comment:
Let's make these into module-level constants. May I suggest names like `REIMPORT_FRESH_FIELDS_ITEM` or something to make it clear where they will be used?